### PR TITLE
[WIP} xbps-triggers: ensure operating in rootdir

### PR DESCRIPTION
**Pull request by Chocimier**
Check others:
- [x] binfmts: broken
- [ ] gconf-schemas
- [x] xml-catalog

### CODE REVIEW

**Comment by ericonr** [comment]
Running the python hooks when installing on another dir at all feels quite wrong, tbh.

Besides that, looks good.

File: srcpkgs/xbps-triggers/files/gtk-icon-cache [hunk 1]
```diff
@@ -23,8 +23,8 @@ targets)
 run)
 	if [ "$TARGET" = "post-remove" -a "${PKGNAME}" = "gtk+" ]; then
 		for dir in ${gtk_iconcache_dirs}; do
-			if [ -f ${dir}/icon-theme.cache ]; then
-				rm -f ${dir}/icon-theme.cache
+			if [ -f .${dir}/icon-theme.cache ]; then
+				rm -f .${dir}/icon-theme.cache
```
Comment (sgn) [inline srcpkgs/xbps-triggers/files/gtk-icon-cache:27]: Would it be better to use `./` instead of `.`?

Comment (Anachron) [reply srcpkgs/xbps-triggers/files/gtk-icon-cache:27]: Why even bother with `./`?

Comment (ericonr) [reply srcpkgs/xbps-triggers/files/gtk-icon-cache:27]: @sgn I believe these are always generated automatically, so we are sure that they will contain a leading slash.

**Comment by Chocimier** [comment]
@ericonr, why do you see it wrong to byte-compile?

**Comment by ericonr** [comment]
@Chocimier I dislike the idea of running binaries for an arch other than the current one; but I guess if you're doing that, you should be using `XBPS_TARGET_ARCH` instead of `ARCH`

**Comment by Chocimier** [comment]
It is not only that triggers try to write to host root. Their executables are linked with dynamic libraries from host, eg. resulting in loud errors when installing other libc, and likely silent breakage when having outdated libraries in host.

Running triggers for alternative rootdir should be disabled at whole, I guess.

**Comment by Chocimier** [comment]
Merging what's here, but even installing into rootdir on same arch as user is broken and needs https://github.com/void-linux/xbps/issues/372 .

**Pull request merged**